### ISSUE TITLE
Building project with completion server not supported?

### ISSUE THREAD

**Issue by b005t3r** [HaxeFoundation/intellij-haxe#572]
I can't find a way of building HXML-based projects using Completion server (which is much faster). The problem is there's no place for additional command line arguments, so I can't call ```haxe --connect PORT``` when building the project.

**Comment by b005t3r** [comment]
OK, actually there's a workaround: I can precede the HXML file path with ```--connect PORT``` and it works (I can't believe how much faster it is this way!), but it would be great to have a build-in feature for it.

**Comment by b005t3r** [comment]
OK, I'm a bit of an idiot - the previous workaround doesn't really work, but simply putting ```--connect PORT``` in the HXML file does work :)

I think I can't close it on my own, sorry for the unnecessary fuss.

**Comment by EricBishton** [comment]
I agree, and it's on my list.

Whether or not you see any speed improvement actually depends upon whether you are using OpenFL or not, and whether you build from inside of the IDE (believe it or not, there are many who don't because the IDE makes a lot of assumptions about the build).  In the OpenFL case, it's because we use `lime --display` to get the set of variables to pass on the Haxe command line; that's different from the actual variables that lime uses to compile.  In the latter case, it's because the cache isn't fast unless it's been pre-warmed, and pre-warming requires an actual build (with `--no-output`).  Just running completions does not populate the cache.

As far as the command line for HXML, putting the `--connect <port>` value in the hxml is not optimal because then you have to remove it to get a real build.  There should probably be a dialog for extra parameters to go on the command line, but then again, you'd have to remove them to get a real build from IDEA.  There are other hacks we could do, but just getting to supporting the server properly is the right solution.

**Comment by b005t3r** [comment]
BTW, is there a difference between a "real build" and a completion server build? It's faster, but the result is still the same, right?

**Comment by EricBishton** [comment]
No, for a 'completion build', the compiler does not actually do any compilation or create any output to the hard drive.  It only gives completion/type info and quits right after parsing (and tries harder not to die because of compile errors).  A 'real build' is one that actually builds/compiles your program, (unless `--no-output` is specified, then it does everything a normal compile would do -- far more than a completion build -- except write the output to disk).

**Comment by EricBishton** [comment]
> As far as the command line for HXML, putting the --connect <port> value in the hxml is not optimal because then you have to remove it to get a real build.

I mis-spoke here.  You do not have to remove the values from the hxml.  You simply have to *always* use the compile server, which makes for faster 'real builds', too.

**Comment by b005t3r** [comment]
All right, understood :)
